public/index.js

```diff
@@ -236,6 +240,20 @@
       });
     }
 
+    function addHaloClickEvent(element){
+      if (!element){ return; }
+      element.addEventListener("click", function(){
+        addHalo(element);
+      });
+    }
+
+    var prev = null;
+    function addHalo(element){
```

Comment:
So in this case you are using outer scope, so it makes sense to keep the function definition inside the forEach.

You should keep the number of these functions to a minimum.

By moving the other functions out, it makes it more explicit when you have a function inside..